### ISSUE TITLE
add php to CI

### ISSUE THREAD

**Issue by pierrejoye** [libgd/libgd#765]
Add build and run php tests to our CI

- 7.4, 8.0 and 8.1 for 2.3
- 8.0 up for 2.4+

**Comment by cmb69** [comment]
I'm not sure whether this is a good idea; at least we shouldn't run that for each commit, but rather  nightly or so. An alternative would be to do that for php-src.

**Comment by pierrejoye** [comment]
I am not sure  :)

My thoughts were to have it run vs released php versions, including rc if
we like to.

Open for scheduled runs but I do like to know if something breaks on
commit, not too much later. :)

**Comment by cmb69** [comment]
Yeah, but there may be a performance issue. Currently our CI is quite fast (a few minutes); building php-src and runnings its gd test suite certainly takes longer. And while php-src might be our most important downstream consumer, there are certainly others as well.

**Comment by pierrejoye** [comment]
it runs in parallel. Let me try with a schedule and see how long it takes.
--disable-all with gd should not be that big hopefully  :)

**Comment by cmb69** [comment]
I don't think this makes sense. We are barely able to get our own CI green, but PHP has a pretty comprehensive CI running well (testing against system libgd on most systems, but using bundled libgd on Windows).

**Comment by vapier** [comment]
i'd be OK with a build test with the latest version of php (maybe even ToT vs latest release) since they typically integrate the source tree directly

**Comment by cmb69** [comment]
> since they typically integrate the source tree directly

If you are referring to the libgd source tree: no, they don't. [PHP has a bundled libgd](https://github.com/php/php-src/tree/master/ext/gd/libgd) which is *somewhat* up-to-date with what we have, but has notable differences in gd_interpolationc. and gdft.c. The bundled libgd is the only option on Windows, but can be used on other systems, too. However, usually other systems use our releases (possibly patched). And PHP has its own GD test suite (currently about 300 test cases).

**Comment by vapier** [comment]
Yes, I meant the bundled copy they carry. That is integrated directly. Maybe not by checking out our source, but it's basically bundled. 

There could be value in seeing if our changes to the latest GD code breaks PHP in unforseen ways. 

But it'd probably require figuring out the divergence between the two source bases first.  I'm not volunteering to sort that out, just agreeing that I think there's value in it.

**Comment by thg2k** [comment]
> But it'd probably require figuring out the divergence between the two source bases first. I'm not volunteering to sort that out, just agreeing that I think there's value in it.

I already did a fair amount of work in this direction, which is what prompted me to open PR #883 

The problem is that there is a sheer amount of noise in the two source trees, so I manually processed each file, line by line, and merged all non-functional changes, so that diffing the resulting trees yields a meaningful diff, showing the actual functional differences. The problem is that some CS issues are better solves on the PHP side and should be improved on the libgd side (you can see a sample in the mentioned PR). So before doing further work on this I wanted to see if that PR can make it into the repo.
